src/bin/io/prompt.rs

```diff
@@ -112,25 +112,30 @@ where
 }
 
 /// Request the user to input the recovery phrase
-pub(crate) fn request_recovery_phrase() -> String {
+pub(crate) fn request_recovery_phrase() -> anyhow::Result<String> {
     // let the user input the recovery phrase
-    let q = Question::input("phrase")
-        .message("Please enter the recovery phrase:")
-        .validate_on_key(|phrase, _| {
-            Mnemonic::from_phrase(phrase, Language::English).is_ok()
-        })
-        .validate(|phrase, _| {
-            if Mnemonic::from_phrase(phrase, Language::English).is_ok() {
-                Ok(())
-            } else {
-                Err("Please enter a valid recovery phrase".to_string())
-            }
-        })
-        .build();
+    let mut attempt = 1;
+    loop {
+        let q = Question::input("phrase")
+            .message("Please enter the recovery phrase:")
+            .build();
+
+        let a = requestty::prompt_one(q).expect("recovery phrase");
+        let phrase = a.as_string().unwrap_or_default();
+
+        match Mnemonic::from_phrase(phrase, Language::English) {
+            Ok(phrase) => break Ok(phrase.to_string()),
 
-    let a = requestty::prompt_one(q).expect("recovery phrase");
-    let phrase = a.as_string().unwrap().to_string();
-    phrase
+            Err(err) if attempt > 2 => match err.downcast_ref::<ErrorKind>()
+            {
+                Some(ErrorKind::InvalidWord) => bail!("Invalid recovery phrase, reached the maximum number of attempts"),
+                _=> return Err(err)}
+            Err(_) => {
+                println!("Invalid recovery phrase please try again");
+                attempt += 1;
+            }
+        }
+    }
```

Comment:
I'm not sure why the formatter didn't fix it, but there are several issue of formatting here (e.g. the space is missing in `_=>`, should have been `_ =>`).

I would also change the error message. My suggestion would be something like the following:
```rust
            Err(err) if attempt > 2 => match err.downcast_ref::<ErrorKind>() {
                Some(ErrorKind::InvalidWord) => {
                    bail!("Reached the maximum number of attempts")
                }

                _ => return Err(err),
            },
```